rom/laghos.cpp

```diff
@@ -954,839 +1056,888 @@ int main(int argc, char *argv[])
+                if (pointRequiringResidual != -1)
+                {
+                    int nearestROM = parameterPointGreedySelector->getNearestROM(pointRequiringResidual);
+                    romOptions.basisIdentifier = "_" + to_string(paramPoints[nearestROM]);
+                    romOptions.blast_energyFactor = paramPoints[pointRequiringResidual];
+                }
+                else
+                {
+                    rom_online = false;
+                }
             }
-        }
 
-        if (visualization)
-        {
-            vis_v->close();
-            vis_e->close();
+            if (!rom_online)
+            {
+                int nextSampleParameterPoint = parameterPointGreedySelector->getNextParameterPoint();
+                if (nextSampleParameterPoint != -1)
+                {
+                    romOptions.basisIdentifier = "_" + to_string(paramPoints[nextSampleParameterPoint]);
+                    romOptions.blast_energyFactor = paramPoints[nextSampleParameterPoint];
+                    rom_offline = true;
+                }
+                else
+                {
+                    parameterPointGreedySelector->writeSampledPoints(outputPath + "/" + "sampled_points");
+                    rom_build_database = false;
+                }
+            }
         }
     }
-
-    totalTimer.Stop();
-    if (mpi.Root()) {
-        if(rom_online) cout << "Elapsed time for online preprocess: " << onlinePreprocessTimer.RealTime() << " sec\n";
-        if(rom_restore) cout << "Elapsed time for restore phase: " << restoreTimer.RealTime() << " sec\n";
-        if(rom_offline) cout << "Elapsed time for sampling in the offline phase: " << samplerTimer.RealTime() << " sec\n";
-        if(rom_offline) cout << "Elapsed time for basis construction in the offline phase: " << basisConstructionTimer.RealTime() << " sec\n";
-        cout << "Elapsed time for time loop: " << timeLoopTimer.RealTime() << " sec\n";
-        cout << "Total time: " << totalTimer.RealTime() << " sec\n";
-    }
-
-    // Free the used memory.
-    if (ode_solver != nullptr) delete ode_solver;
-    if (pmesh != nullptr) delete pmesh;
-    if (oper != nullptr) delete oper;
-    if (rho != nullptr) delete rho;
-    if (mat_fec != nullptr) delete mat_fec;
-    if (mat_fes != nullptr) delete mat_fes;
-    if (mat_gf != nullptr) delete mat_gf;
-    if (mat_gf_coeff != nullptr) delete mat_gf_coeff;
-    if (L2FESpace != nullptr) delete L2FESpace;
-    if (H1FESpace != nullptr) delete H1FESpace;
-    if (S != nullptr) delete S;
-    if (S_old != nullptr) delete S_old;
-    if (x_gf != nullptr) delete x_gf;
-    if (v_gf != nullptr) delete v_gf;
-    if (e_gf != nullptr) delete e_gf;
-    if (rho_gf != nullptr) delete rho_gf;
-    if (vis_rho != nullptr) delete vis_rho;
-    if (vis_v != nullptr) delete vis_v;
-    if (vis_e != nullptr) delete vis_e;
-    if (visit_dc != nullptr) delete visit_dc;
-    if (v_coeff != nullptr) delete v_coeff;
```

Comment:
It seems some minor changes to this file resulted in minor formatting changes to most of the lines, making it almost impossible to review. @kevinhkhuynh could you please summarize what real changes were made in this file?

Reply:
Removing the do while loop has reverted the formatting changes.